libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -382,7 +380,8 @@ method subscribePeer*(p: PubSub, peer: PeerId) {.base, gcsafe.} =
   ## messages
   ##
 
-  discard p.getOrCreatePeer(peer, p.codecs)
+  let pubSubPeer = p.getOrCreatePeer(peer, p.codecs)
```

Comment:
This creates a remote peer `pubsubpeer` but with the local peer `p.codes` and adds the remote peer to the local peer's peers table. When this executes https://github.com/vacp2p/nim-libp2p/blob/master/libp2p/protocols/pubsub/pubsub.nim#L379 the remote peer continues with the wrong codecs, as it's already present in the table.